File: 알고리즘(Algorithms)/1. 정렬(Sorting)/Insertion sort/Insertion sort.cpp
#include <iostream>

int main(void)
{
	int arr[20] = {1,2,3,4,5,6,6,1,2,333,4,3,2,5,6,7,0,11,23,40 };

	int temp; //임시변수

	for (int i = 1; i < 20; i++){
		temp = arr[i];
		for (int j = i - 1; j >= 0; j--){
			if (arr[j]>temp)
				std::swap(arr[j], arr[j+1]);
		}
	}

	for (int i = 0; i < 20; i++) std::cout << arr[i] << " ";

	return 0;
}

File: 알고리즘(Algorithms)/1. 정렬(Sorting)/Insertion sort/InsertionSort.java
package com.haram;

public class InsertionSort {

    public static void main(String[] args) {
        int arr[] = {5,4,9,0,1,2,3,8,11,12};

        for(int i=0; i<9; i++) {
            for(int j=i+1; j>0; j--){
                if(arr[j]<arr[j-1])
                    swap(arr, j, j-1);
            }
        }

        for(int i=0; i<10; i++) System.out.print(arr[i] + " ");
    }

    public static void swap(int arr[], int index1, int index2) {
        int temp = arr[index1];
        arr[index1] = arr[index2];
        arr[index2] = temp;
    }
}
